src/components/Searchbar/Searchbar.js

```diff
@@ -0,0 +1,115 @@
+import { gray } from "d3-color";
+import { useRouter } from "next/router";
+import React, { useEffect, useState } from "react";
+import AsyncSelect from "react-select/async";
+
+const searchBarStyles = {
+  control: (provided) => ({
+    ...provided,
+    fontFamily: "Gotham-Book",
+    color: "white",
+    backgroundColor: gray,
+    fontSize: 16,
+    border: "0px",
+    borderRadius: "8px",
+    padding: "4px",
+  }),
+  option: (provided, state) => ({
+    ...provided,
+    fontFamily: "Gotham-Book",
+    fontSize: 14,
+    color: "black",
+  }),
+  menu: (styles) => ({
+    ...styles,
+    border: "0px",
+    borderRadius: "8px",
+  }),
+  input: (provided) => ({
+    ...provided,
+    color: "white",
+  }),
+  placeholder: (provided, { isFocused }) => ({
+    ...provided,
+    color: isFocused ? "gray" : "white",
+  }),
+};
+
+//search component for search page
+function Searchbar() {
+  const router = useRouter();
+  const [searchInput, setSearchInput] = useState("");
+  const [selectedValue, setSelectedValue] = useState(null);
+
+  //helper function to fetch from api
+  function fetchData(url) {
+    return new Promise((resolve, reject) => {
+      fetch(url)
+        .then((res) => {
+          if (!res.ok) {
+            reject(res);
+          }
+          resolve(res.json());
+        })
+        .catch(reject);
+    });
+  }
+
+  //helper functions to concat strings
+  function clanSearchUrl(input) {
+    return "/api/search?q=" + input + "&type=clan&limit=2";
+  }
+  function matchSearchUrl(input) {
+    return "api/matches?match_id=" + input + "&limit=3&sort_by=date&desc=true"; //for now: limit to max. 2 matches
+    //no sorting yet
+  }
+
+  //function to load clan and match options from API
+  async function loadOptions(input, callback) {
+    const resClan = await fetchData(clanSearchUrl(input));
+    const resMatch = await fetchData(matchSearchUrl(input));
+
+    const clanoptions = resClan.map((i) => ({
+      label: i.name,
+      value: i.tag,
+      type: "clan",
+    }));
+    const matchoptions = resMatch.map((i) => ({
+      label: i.match_id,
+      value: i.match_id,
+      type: "match",
+    }));
+
+    callback([].concat(clanoptions, matchoptions));
```

Comment:
```suggestion
    callback(clanoptions.concat(matchoptions));
```
```suggestion
    callback([...clanoptions, ...matchoptions]);
```

you can either concat them directly or use the spread operator